src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/IncludeFlagUtils.cs

```diff
@@ -90,7 +90,7 @@ internal static class IncludeFlagUtils
             // Queue all direct references
             foreach (var graph in targetGraph.Graphs)
             {
-                foreach (var root in graph.InnerNodes)
+                foreach (var root in graph.InnerNodes.Where(n => !n.Item.IsCentralTransitive))
```

Comment:
Why are we skipping this?

Reply:
To queue only direct dependencies. An item that has this flag is not a direct defined by project dependency.

Reply:
I imagine this is not a req introduced in this PR, but a bug fix correct?

Reply:
The previous  behavior is to queue only the direct references nodes. There is a comment above at line 90 as well. This only exclude the nodes that are centralTransitive.

Reply:
Yes, correct.